src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -1009,6 +1009,47 @@ private static bool TryReplaceWithAlias(this ExpressionSyntax node, SemanticMode
                 return false;
             }
 
+            // Avoid the TryReplaceWithAlias algorithm if the tree has no using alias directives. Since the input node
+            // might be a speculative node (not fully rooted in a tree), we use the position of the evaluation to find
+            // the equivalent node in the original tree, and from there determine if the tree has any using alias
+            // directives.
+            if (semanticModel.SyntaxTree.TryGetRoot(out var root))
```

Comment:
Why the choice of TryGetRoot here? It's a pretty "special" method that we generally point people away from unless you _really_ need it. (Think GC.KeepAlive.) If the assumption is that the root is going to be held because of ExpressionSyntax, I think it might still look less scary to just call GetRoot() since it'll fast-path either way?

Reply:
(TryGet* methods on these things are pretty much automatic-code-review questioning from me, a la GC.KeepAlive.)

Reply:
➡️ Documented this and the performance constraint in recent updates.